doc/design/refactor_discuess/readme.md

```diff
@@ -0,0 +1,273 @@
+# Paddle 计算重构设计文档
+
+整体设计如下图所示:
+
+![alt](./pics/whole_archi.png)
+
+主要的概念包括:
+
+* Workspace: 全部的神经网络层的参数和输入，均由一个全局对象所管理。内存/显存的申请，释放，resize都使用该对象管理。
+	* 内存申请后的类型是`TensorBuffer`，该类型是一个`void*`指针，size
```

Comment:
`typedef void * TensorBuffer` looks an unacceptable over-simplification.

Neither TensorFlow's tensor nor MXNet's Blob is a `void*`.

TensorFlow's Tensor has the following data members:

```cpp
 TensorShape shape_;
 TensorBuffer* buf_;
```

where  [`TensorBuffer`](http://yangff.coding.me/tf-doc/codebrowser/codebrowser/tensorflow/core/framework/tensor.h.html#tensorflow::TensorBuffer) is an interface derives from [`RefCounted`](http://yangff.coding.me/tf-doc/codebrowser/codebrowser/tensorflow/core/lib/core/refcount.h.html#tensorflow::core::RefCounted).